server/src/main/java/org/apache/iotdb/db/qp/logical/crud/InOperator.java

```diff
@@ -0,0 +1,213 @@
+    return values;
+  }
+
+  @Override
+  public void reverseFunc() {
+    not = !not;
+  }
+
+  @Override
+  protected Pair<IUnaryExpression, String> transformToSingleQueryFilter(
+      IQueryProcessExecutor executor) throws LogicalOperatorException, PathException {
+    TSDataType type = executor.getSeriesType(singlePath);
+    if (type == null) {
+      throw new PathException(
+          "given seriesPath:{" + singlePath.getFullPath() + "} don't exist in metadata");
+    }
+    IUnaryExpression ret;
+
+    switch (type) {
+      case INT32:
+        Set<Integer> integerValues = new HashSet<>();
+        for (String val : values) {
+          integerValues.add(Integer.valueOf(val));
+        }
+        ret = In.getUnaryExpression(singlePath, integerValues, not, valueToString);
+        break;
+      case INT64:
+        Set<Long> longValues = new HashSet<>();
+        for (String val : values) {
+          longValues.add(Long.valueOf(val));
+        }
+        ret = In.getUnaryExpression(singlePath, longValues, not, valueToString);
+        break;
+      case BOOLEAN:
+        Set<Boolean> booleanValues = new HashSet<>();
+        for (String val : values) {
+          booleanValues.add(Boolean.valueOf(val));
+        }
+        ret = In.getUnaryExpression(singlePath, booleanValues, not, valueToString);
+        break;
+      case FLOAT:
+        Set<Float> floatValues = new HashSet<>();
+        for (String val : values) {
+          floatValues.add(Float.parseFloat(val));
+        }
+        ret = In.getUnaryExpression(singlePath, floatValues, not, valueToString);
+        break;
+      case DOUBLE:
+        Set<Double> doubleValues = new HashSet<>();
+        for (String val : values) {
+          doubleValues.add(Double.parseDouble(val));
+        }
+        ret = In.getUnaryExpression(singlePath, doubleValues, not, valueToString);
+        break;
+      case TEXT:
+        Set<Binary> binaryValues = new HashSet<>();
+        for (String val : values) {
+          binaryValues.add(
+              (val.startsWith("'") && val.endsWith("'")) || (val.startsWith("\"") && val
+                  .endsWith("\"")) ? new Binary(val.substring(1, val.length() - 1))
+                  : new Binary(val));
+        }
+        ret = In.getUnaryExpression(singlePath, binaryValues, not, valueToString);
+        break;
+      default:
+        throw new LogicalOperatorException(type.toString(), "");
+    }
+
+    return new Pair<>(ret, singlePath.getFullPath());
+  }
+
+  @Override
+  public String showTree(int spaceNum) {
+    StringContainer sc = new StringContainer();
+    for (int i = 0; i < spaceNum; i++) {
+      sc.addTail("  ");
+    }
+    sc.addTail(singlePath.toString(), this.tokenSymbol, not, valueToString, ", single\n");
+    return sc.toString();
+  }
+
+  @Override
+  public InOperator clone() {
+    InOperator ret;
+    try {
+      ret = new InOperator(this.tokenIntType, singlePath.clone(), not, new HashSet<>(values));
```

Comment:
return null isn't a good idea, It is really necessary to use try-catch and log in here?

Reply:
fixed

Reply:
can you remove log in this object?

Reply:
fixed.